### ISSUE TITLE
[a11y]: Multiple search landmarks have identical aria-labels causing accessibility violation

### ISSUE THREAD

**Issue by ss-anju** [carbon-design-system/carbon#21244]
## Environment

> Operating system
> macOS

> Browser
> Chrome

> Automated testing tool and ruleset
>[IBM Equal Access Accessibility Checker](https://github.com/IBMa/equal-access/tree/main-4.x) (ruleset ["4.0.9"](https://github.com/IBMa/equal-access/blob/main-4.x/rule-server/src/static/archives.json))

> Assistive technology used to verify
> Screen Reader

## Detailed description

> What version of the Carbon Design System are you using?
> v11 

> What did you expect to happen?
>When multiple Search or ExpandableSearch components are present on the same page, each search landmark should have a unique accessible name that distinguishes it from others. This allows screen reader users to identify which search field they're interacting with when navigating by landmarks.

> What happened instead?
>All Search/ExpandableSearch components on the same page have identical aria-label values on their search landmark regions, making them indistinguishable to assistive technology users.
>Root cause: The Search component hardcodes the search landmark's aria-label to use the placeholder prop value.

> What WCAG 2.1 checkpoint does the issue violate?
[ 1.3.1 Info and Relationships](https://www.ibm.com/able/requirements/requirements/?version=v7_3#1_3_1)
[2.4.1: Bypass Blocks - Landmarks must be uniquely identifiable](https://www.ibm.com/able/requirements/requirements/?version=v7_3#2_4_1)
ARIA Rule ID: aria_landmark_name_unique

## Steps to reproduce the issue

1. Open [this CodeSandbox example.](https://codesandbox.io/p/sandbox/a11y-search-295wfh?file=%2Fsrc%2Findex.js)
2. Run the [IBM A11Y Checker V 4.0.9](https://www.ibm.com/able/toolkit/tools/)
3. View the Violation Report.

> CodeSandbox example
> https://codesandbox.io/p/sandbox/a11y-search-295wfh?file=%2Fsrc%2Findex.js

## Additional information

- Screenshot

<img width="1728" height="963" alt="Image" src="https://github.com/user-attachments/assets/3008fca4-7ee3-4dcd-b913-c841b0c00136" />

**Comment by sunny-kukkar-ibm** [comment]
Kindly use "@carbon/react" as "carbon-components-react" is already deprecated. Search component in "@carbon/react" aria-label can be giving with "placeholder" prop. 

Here is an example:
https://stackblitz.com/edit/4zy15jj2?file=src%2FApp.jsx

https://react.carbondesignsystem.com/?path=/docs/components-search--overview
